dimod/constrained.py

```diff
@@ -681,10 +685,21 @@ def add_variable(self, v: Variable, vartype: VartypeLike,
         Examples:
             >>> from dimod import ConstrainedQuadraticModel, Integer
             >>> cqm = ConstrainedQuadraticModel()
-            >>> cqm.add_variable('i', 'INTEGER')
+            >>> cqm.add_variable(INTEGER', 'i')
             'i'
 
         """
+        try:
+            vartype = as_vartype(vartype, extended=True)
+        except TypeError:
```

Comment:
This seems to assume that any `TypeError` is caused by a reversed order. But maybe the order is correct but the type is bad, for example `dimod.Vartype.CONTINUOUS`

Reply:
In that case the user would get an attribute error.

But even in the `cqm.add_variable('CONTINUUOS', 'i')` case, the user will still get an error message. It will likely be slightly confusing but from the combo of warning and exception they should be able to figure out what happened.